store/tikv/2pc.go

```diff
@@ -581,6 +591,64 @@ func (c *twoPhaseCommitter) prewriteSingleBatch(bo *Backoffer, batch batchKeys)
 	}
 }
 
+type ttlManager struct {
+	// 0 uninitialized, 1 running, 2 closed
+	initialized uint32
+	ch          chan struct{}
+}
+
+func (tm *ttlManager) run(c *twoPhaseCommitter) {
+	// Run only once.
+	if !atomic.CompareAndSwapUint32(&tm.initialized, 0, 1) {
+		return
+	}
+	go tm.keepAlive(c)
+}
+
+func (tm *ttlManager) close() {
+	if !atomic.CompareAndSwapUint32(&tm.initialized, 1, 2) {
+		return
+	}
+	close(tm.ch)
+}
+
+func (tm *ttlManager) keepAlive(c *twoPhaseCommitter) {
+	bo := NewBackoffer(context.Background(), pessimisticLockMaxBackoff)
+	// PessimisticLockTTL is 15s in conf, so the ticker is 5s.
+	// In the test, PessimisticLockTTL is set to 300ms, then ticker is 100ms.
+	ticker := time.NewTicker(time.Duration(PessimisticLockTTL) * time.Millisecond / 3)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-tm.ch:
+			return
+		case <-ticker.C:
+
+			now, err := c.store.GetOracle().GetTimestamp(bo.ctx)
+			if err != nil {
+				err1 := bo.Backoff(BoPDRPC, err)
```

Comment:
Is it better to create backoffer for each tick? Now it seems the longer it runs, the more possibly the request fails. I'm afraid it may make long transactions likely to fail.

Reply:
If err == nil, the Backoffer would not consume its `maxSleep`, i.e. the Backoffer would not consume its `maxSleep` in the normal ticker.

Reply:
It may be consumed by an accidental network jitter, or a leader change on tikv.